src/ruby/lib/grpc/core/call.rb

```diff
@@ -0,0 +1,20 @@
+module GRPC
+  module Core
+    class Call
+      # call-seq:
+      # metadata = call.metadata
+      #
+      # Gets the metadata object saved the call.
+      attr_reader :metadata
+
+      # call-seq:
+      # call.metadata = metadata
+      #
+      # Saves the metadata hash on the call.
+      def metadata=(val)
```

Comment:
This PR is just to convert C to Ruby, as pure Ruby should be a little easier to work on.  It made no changes in the API surface area.  If the intention is that the `metatdata=` method is not public API, it should probably be a private method or at the very least `nodoc`.